osf/management/commands/file_view_counts.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import unicode_literals
+import logging
+
+import django
+django.setup()
+import json
+import requests
+import urllib
+
+
+from django.core.management.base import BaseCommand
+from django.db import transaction
+from osf.models import BaseFileNode, PageCounter
+from website import settings
+
+from scripts import utils as script_utils
+
+logger = logging.getLogger(__name__)
+
+
+def set_file_view_counts(state, *args, **kwargs):
+    # get all osfstorage files which is_deleted == False
+    files = BaseFileNode.resolve_class('osfstorage', BaseFileNode.FILE).active.all()
+
+    for file_node in files:
+        # for each file get the file view counts from keen
+        query = [{'property_name': 'page.info.path', 'operator': 'eq', 'property_value': file_node._id}]
+
+        query = urllib.quote(json.dumps(query))
+        url = 'https://api.keen.io/3.0/projects/{}/queries/count' \
```

Comment:
@felliott Given the recent keen changes, is this the approach we should take, or would it be better to try to do some kind of batching? I'm not sure if batching is possible/convenient, so this is probably fine, but wanted you opinion.

Reply:
Keen.io Compute Pricing Guide: https://keen.io/docs/compute/compute-pricing-guide/

Compute is charged at $0.01/1M properties scanned.  Number of properties scanned is euqal to E * P, where E is number of events in the timeframe and P is number of properties needed to calculate the query (see link for details).

Private pageviews has about 350k events over the last few days.  P for this query is 2: 1 + one filter property.  That means executing one of these queries involves 350k * 2 or 700k properties scanned, which would cost about $0.007.  But to calculate total cost, we'd have to multiply by the number of time we issued it, so it would end up at `$0.007 * len(files)`.

If instead you did a `group_by` on `page.info.path` that would return the counts for all files id (and non file-id pages) in one query for $0.007.  That might run into payload limits, though.

The third option is to batch, as @brianjgeiger said.  You'd have to futz around a bit to figure out the largest batch size that wouldn't hit payload limits, but that would probably be cheaper than this approach, too.  You'd change the filter parameter to be a `'page.info.path IN ['mst3k', 'ezcas', 'moooo', 'b33fy', ...]` and add a group by.  That would bring the P up to 3, so each query would be $0.012, but you could divide the number of queries you issue by batch size.  See: https://keen.io/docs/compute/query-tuning/#calculating-metrics-for-many-users-at-once

You'll need to decide what the right approach is based of how often you expect to run this and how big `len(files)` is.